jax/_src/util.py

```diff
@@ -392,3 +393,22 @@ def maybe_named_axis(axis, if_pos, if_named):
   except TypeError:
     named = True
   return if_named(axis) if named else if_pos(pos)
+
+def distributed_debug_log(*pairs):
+  """Format and log `pairs` if config.jax_distributed_debug is enabled.
+
+  Args:
+    pairs: A sequence of label/value pairs to log. The first pair is treated as
+    a heading for subsequent pairs.
+  """
+  if config.jax_distributed_debug:
+    msg = "\nDISTRIBUTED_DEBUG_BEGIN"
+    try:
+      msg += f"\n{pairs[0][0]}: {pairs[0][1]}"
+      for label, value in pairs[1:]:
+        msg += f"\n  {label}: {value}"
```

Comment:
This technically goes against the usual performance best-practices for formatting strings in Python, but I'm not sure how performance sensitive this piece of code is. In any case, I'd switch it to appending to a list, followed by a `'\n'.join(lines)` to make it O(n) instead of O(n^2).

Reply:
Ah good catch. I was thinking that perf didn't matter, but it might, and it's good to do this regardless.